spring-cloud-commons/src/main/java/org/springframework/cloud/client/serviceregistry/AbstractAutoServiceRegistration.java

```diff
@@ -76,6 +82,32 @@ protected AbstractAutoServiceRegistration(ServiceRegistry<R> serviceRegistry,
 		this.properties = properties;
 	}
 
+	protected AbstractAutoServiceRegistration(ServiceRegistry<R> serviceRegistry,
+			AutoServiceRegistrationProperties properties,
+			List<RegistrationManagementLifecycle<R>> registrationManagementLifecycles,
+			List<RegistrationLifecycle<R>> registrationLifecycles) {
+		this.serviceRegistry = serviceRegistry;
+		this.properties = properties;
+		this.registrationManagementLifecycles = registrationManagementLifecycles;
+		this.registrationLifecycles = registrationLifecycles;
+	}
+
+	protected AbstractAutoServiceRegistration(ServiceRegistry<R> serviceRegistry,
+			AutoServiceRegistrationProperties properties, List<RegistrationLifecycle<R>> registrationLifecycles) {
+		this.serviceRegistry = serviceRegistry;
+		this.properties = properties;
+		this.registrationLifecycles = registrationLifecycles;
+	}
+
+	public void addRegistrationManagementLifecycles(
+			List<RegistrationManagementLifecycle<R>> registrationManagementLifecycles) {
+		this.registrationManagementLifecycles.addAll(registrationManagementLifecycles);
+	}
+
+	public void addRegistrationLifecycles(List<RegistrationLifecycle<R>> registrationLifecycles) {
```

Comment:
Is this needed? Is there a reason all the `RegistrationLifecycle` objects can't be initialised in the constructor and the field storing them cannot be `final`?